### PR TITLE
fix: Validate asset config values are URLs

### LINKED ISSUE #164: Validate asset configs to use https:// addresses
Asset-related configuration fields, such as `logo_asset` and `favicon_asset` **need** to be URLs.

Using a non-HTTP or HTTPS url will cause Rails to attempt to find the asset in its local asset pipeline (app/assets/images/...), which will fail, and take down the website. This also takes down the management UI, which makes fixing the error even more difficult.

Might be easiest to solve this by setting some sort of URL validator for any config field that includes `_asset` in the name

### PR DESCRIPTION
**Pull request by dquach217**
Fixes #164

### CODE REVIEW

**Comment by dquach217** [comment]
@sman591 not sure how to go about reducing the code complexity

File: app/controllers/manage/configs_controller.rb [hunk 1]
```diff
@@ -17,7 +17,9 @@ def update
     value = params[:hackathon_config][key]
     value = true if value == "true"
     value = false if value == "false"
-    if @config.value != value
+    if @config.var.end_with?("_asset") && !value.start_with?('http://', 'https://')
+      redirect_to manage_configs_path, notice: "Config \"#{key}\" was not changed"
```
Comment (sman591) [inline app/controllers/manage/configs_controller.rb]:
```suggestion
      flash[:alert] = "Config \"#{key}\" must start with http:// or https://"
      render :edit
```

This should help it be more clear why the value was rejected, and using `render :edit` will render the same page (hopefully) with the same value the user tried to use

File: test/controllers/manage/configs_controller_test.rb [hunk 2]
```diff
@@ -115,6 +115,18 @@ class Manage::ConfigsControllerTest < ActionController::TestCase
...
+    should "update logo_asset with a url" do
+      HackathonConfig["logo_asset"] = ''
+      patch :update, params: { id: "logo_asset", hackathon_config: { logo_asset: "https://picsum.photos/200" } }
+      assert_equal "https://picsum.photos/200", HackathonConfig["logo_asset"]
+    end
+
+    should "update logo_asset with an asset that is not URL based" do
+      HackathonConfig["logo_asset"] = ''
+      patch :update, params: { id: "logo_asset", hackathon_config: { logo_asset: "test" } }
+      assert_equal '', HackathonConfig["logo_asset"]
```
Comment (sman591) [inline test/controllers/manage/configs_controller_test.rb]: Might be good to add:

```ruby
assert_redirected_to edit_manage_config_path("logo_asset")
```

**Comment by dquach217** [comment]
@sman591 Change has been made and added a bit more

File: test/controllers/manage/configs_controller_test.rb [hunk 3]
```diff
@@ -115,6 +115,20 @@ class Manage::ConfigsControllerTest < ActionController::TestCase
...
     end
 
+    should "update logo_asset with a url" do
+      HackathonConfig["logo_asset"] = ''
+      patch :update, params: { id: "logo_asset", hackathon_config: { logo_asset: "https://picsum.photos/200" } }
+      assert_equal "https://picsum.photos/200", HackathonConfig["logo_asset"]
+      assert_redirected_to manage_configs_path
+    end
+
+    should "update logo_asset with an asset that is not URL based" do
```
Comment (sman591) [inline test/controllers/manage/configs_controller_test.rb]:
```suggestion
    should "not update logo_asset with an asset that is not URL based" do
```

**Pull request merged**